tests/state_test.go

```diff
@@ -238,17 +239,36 @@ func runBenchmark(b *testing.B, t *StateTest) {
 			sender := vm.NewContract(vm.AccountRef(msg.From()), vm.AccountRef(msg.From()),
 				nil, 0)
 
+			var gasUsed uint64
+			var elapsed uint64
+
 			b.ResetTimer()
 			for n := 0; n < b.N; n++ {
-				// Execute the message.
 				snapshot := statedb.Snapshot()
-				_, _, err = evm.Call(sender, *msg.To(), msg.Data(), msg.Gas(), msg.Value())
+
+				b.StartTimer()
+				start := time.Now()
+
+				// Execute the message.
+				_, leftOverGas, err := evm.Call(sender, *msg.To(), msg.Data(), msg.Gas(), msg.Value())
 				if err != nil {
 					b.Error(err)
 					return
 				}
+
+				b.StopTimer()
+				elapsed += uint64(time.Since(start))
+				gasUsed += msg.Gas() - leftOverGas
+
 				statedb.RevertToSnapshot(snapshot)
 			}
+
+			if elapsed < 1 {
+				elapsed = 1
+			}
+			// Keep it as uint64, multiply 100 to get two digit float later
+			mgasps := (100 * 1000 * gasUsed) / elapsed
```

Comment:
Thought: could either the aggregation of gasUsed go above uint64Max, or, alternatively, 100*1000* mult here might push it over ? 
If so, we would be better off keeping track of it as a big.Int.

Reply:
it seems fine:
```bash
➜  go-ethereum git:(benchmark-evm-mgas) fd '.json' tests/evm-benchmarks/benchmarks | # Get all test file paths
xargs jq -r 'values[] | .env.currentGasLimit' | # Extract gas limit
xargs -L 1 cast --to-dec | # Hex -> dec
sort -u | # Unique 
awk '{ print 18446744073709551615 - 100 * 1000 * $0 * 30000 }' # math.MaxUint64 - 100 * 1000 * gasUsed * b.N

15446744073709551616
```